### PR TITLE
Enterprise logo: Resize and crop logo instead of just resizing --> logo should fit into a square

### LINKED ISSUE #9626: Crop the enterprise logo instead of squeezing it to size
## Description
**- As a:** enterprise or admin user
**- On page:** `/admin/enterprises/release-farm/edit#!#images_panel`
**- I want to be able to do:** 

If an enterprise logo is uploaded in a non-square ratio, the image will be squeezed to size. This can lead to unreadable and ugly logos. Cropping at the center (like we do on product images as well) makes sure that at least any text on the image is still readable.

grafik

## Acceptance Criteria & Tests

1. Log in as enterprise
2. Upload a non square logo
3. See the logo is cropped and not squeazed

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

- Closes #9626

#### What should we test?

- As an admin, change enterprise logo in `/admin/enterprises/[ENTERPRISE_PERMALINK]/edit#!#images_panel`
- Use a non square image, and upload it.
- Displayed image should fit a 100*100 square ; image should be cropped at its center
Capture d’écran 2022-11-08 à 14 58 56

- As a shopper, visit the enterpise shopfront. Image should be cropped and 100*100 with right ratio:
Capture d’écran 2022-11-08 à 14 58 08

#### Release notes

Changelog Category: User facing changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/views/admin/enterprises/form/_images.html.haml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     %br
     100 x 100 pixels
   .omega.eight.columns
-    %img{ class: 'image-field-group__preview-image', ng: { src: '{{ Enterprise.logo.medium }}', if: 'Enterprise.logo' } }
+    %img{ class: 'image-field-group__preview-image', ng: { src: '{{ Enterprise.logo.thumb }}', if: 'Enterprise.logo' } }
```
Comment (dacook) [inline app/views/admin/enterprises/form/_images.html.haml:7]: 👌

File: app/models/enterprise.rb [hunk 2]
```diff
@@ -6,9 +6,9 @@ class Enterprise < ApplicationRecord
...
   LOGO_SIZES = {
-    thumb: { resize_to_limit: [100, 100] },
-    small: { resize_to_limit: [180, 180] },
-    medium: { resize_to_limit: [300, 300] },
+    thumb: { gravity: "Center", resize: "100x100^", crop: '100x100+0+0' },
+    small: { gravity: "Center", resize: "180x180^", crop: '180x180+0+0' },
+    medium: { gravity: "Center", resize: "300x300^", crop: '300x300+0+0' },
   }.freeze
   PROMO_IMAGE_SIZES = {
     thumb: { resize_to_limit: [100, 100] },
```
Comment (dacook) [inline app/models/enterprise.rb:14]: Would it be relevant to update the promo images too? Or is that scope creep...

**Review by dacook — COMMENTED** [review]
Thanks for figuring out the ActiveStorage parameters! 
Just wondering if it would be easy to update the next section too, while we're at it?

File: app/models/enterprise.rb (hunk 2 above)
Comment (jibees) [reply app/models/enterprise.rb:14]: Yes, actually, why not. 
@openfoodfoundation/train-drivers-product-owners Should I also resize/crop (if needed) the promo image (like the logo image) to the right size?

Comment (RachL) [reply app/models/enterprise.rb:14]: This issue is not a priority. I would move forward with this PR as is.

Comment (jibees) [reply app/models/enterprise.rb:14]: Moving to `Test Ready` then

**Comment by drummer83** [comment]
Hi @jibees,
Thanks for working on this one!

### Before the PR
I confirmed that non-square logo images are still squeezed to size before your PR. 
In the /admin area they are displayed non-sqeezed and non-square:
image

In the shopfront they are displayed as squeezed to square:
image

### After the PR
After staging the PR I had a look again.
Even without re-uploading the image it is now displayed cropped to size in the shopfront. :arrow_right: This means that all currently squeezed images will be non-squeezed from now on. I think that's good! :heavy_check_mark: 
image

Uploading the image again it is now displayed cropped to a square in the /admin area as well. :arrow_right: This is a proper preview of the image now! :heavy_check_mark: 
image

As well as in the shopfront: :heavy_check_mark: 
image

And in the enterprise modal: :heavy_check_mark: 
image

### Result
I think this one is great improvement and it is ready to go! :muscle: 
Thanks again!
